extensions/jira-search/src/issue.ts

```diff
@@ -79,13 +81,31 @@ function buildJql(query: string): string {
   return jql + " order by lastViewed desc"
 }
 
-function jqlFor(query: string): string {
-  return isIssueKey(query) ? `key=${query}` : buildJql(query)
+function jqlFor(query: string, filter?: IssueFilter): string {
+  const jql = isIssueKey(query) ? `key=${query}` : buildJql(query)
+
+  if (!filter || filter === "allIssues") return jql
+  let extraJqlForFilter = ""
+
+  if (filter === "myIssues") {
+    extraJqlForFilter = "assignee=currentUser()"
+  }
+
+  const extraOperator = query ? "AND" : ""
+
+  return `${extraJqlForFilter} ${extraOperator} ${jql}`
 }
 
-export async function searchIssues(query: string): Promise<ResultItem[]> {
-  const jql = jqlFor(query)
+const isIssueFilter = (filter?: string): filter is IssueFilter => {
+  const filters: readonly string[] = issueFilters
+  return !!(filter && (filters.includes(filter) || filters.includes(filter)))
+}
+
+export async function searchIssues(query: string, filter?: string): Promise<ResultItem[]> {
```

Comment:
Hm, why can‘t we type the `filter` argument here more precisely. You‘ve provided the `SearchCommand` a generic parameter, so that this should be possible or am I missing something. This should completely eliminate the `isIssueFilter()` typeguard.